functests/service_account_creation_test.go

```diff
@@ -0,0 +1,49 @@
+package functests
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var _ = Describe("Service Account Creation", func() {
+	var k8sClient *kubernetes.Clientset
+
+	BeforeEach(func() {
+		RegisterFailHandler(Fail)
+
+		deployManager, err := NewDeployManager()
+		Expect(err).To(BeNil())
+		k8sClient = deployManager.GetK8sClient()
+	})
+
+	Describe("Service Account", func() {
+		serviceAccount := &k8sv1.ServiceAccount{
+			TypeMeta:   metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+			ObjectMeta: metav1.ObjectMeta{Name: operatorName},
+		}
+
+		AfterEach(func() {
+			err := k8sClient.CoreV1().ServiceAccounts("default").Delete(serviceAccount.Name, &metav1.DeleteOptions{})
+			if err != nil && !errors.IsNotFound(err) {
+				Expect(err).To(BeNil())
+			}
+		})
+
+		Context("Create service account", func() {
+			It("and delete service account", func() {
+				By("Creating Service Account")
+				_, err := k8sClient.CoreV1().ServiceAccounts("default").Create(serviceAccount)
+				Expect(err).To(BeNil())
+
+				By("Deleting Service Account")
```

Comment:
There's already a delete in line 30. Is this needed?

Reply:
No, not needed. I have now even made the code simpler. May need BeforeEach and AfterEach later but have removed for now.